tests/system-broker/tests/auth_test.go

```diff
@@ -95,6 +115,66 @@ func TestSystemBrokerAuthentication(t *testing.T) {
 	})
 }
 
+func TestCallingORDServiceWithCert(t *testing.T) {
+	ORDContext, err := pkg.NewTestContext(config)
+	require.NoError(t, err)
+
+	logrus.Infof("registering runtime with name: %s, within tenant: %s", runtimeInput.Name, ORDContext.Tenant)
+	runtime := director.RegisterRuntimeFromInputWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, runtimeInput)
+	defer director.UnregisterRuntimeWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, runtime.ID)
+
+	app, err := director.RegisterApplicationWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, applicationInput)
+	require.NoError(t, err)
+	defer director.UnregisterApplication(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, app.ID)
+
+	securedClient, configuration, certChain := getSecuredClientByContext(t, testCtx, runtime.ID)
+
+	bundle := director.CreateBundle(t, ORDContext.Context, ORDContext.DexGraphqlClient, app.ID, ORDContext.Tenant, testBundleName)
+	api := director.AddAPIToBundleWithInput(t, ORDContext.Context, ORDContext.DexGraphqlClient, bundle.ID, ORDContext.Tenant, apiDefInput)
+
+	t.Run("Should succeed calling ORD service with the same cert used for calling catalog", func(t *testing.T) {
+		req := createCatalogRequest(t)
+		resp, err := securedClient.Do(req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, resp.StatusCode, http.StatusOK)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Contains(t, string(body), "services")
+
+		reqORD := createORDServiceRequest(t, ORDContext, api.ID, api.Spec.ID)
+		respORD, err := securedClient.Do(reqORD)
+
+		require.NoError(t, err)
+		require.NotNil(t, respORD)
+		require.Equal(t, http.StatusOK, respORD.StatusCode)
+	})
+
+	t.Run("Should fail calling ORD service with revoked cert", func(t *testing.T) {
```

Comment:
Is it possible that this test runs before the previous one (the happy path)? If it is possible then this would break the happy path test and we should probably create a dedicated secured client for each test and defer the revocation.

Reply:
It looked to me that subtests are run sequentially because they are simply method invocations (and will get executed in the same order in which they appear in code). [This answer](https://stackoverflow.com/a/39734200/9986012) seems to support that. Also, I have run all tests multiple times now and had never noticed a different execution order.